extensions/amp-wpm-player/0.1/amp-wpm-player.js

```diff
@@ -0,0 +1,557 @@
+
+    /** @private {string} */
+    this.playingState_ = PlayingStates.PAUSED;
+
+    /** @private {!Array<function>} */
+    this.messageListeners_ = [];
+
+    /** @private {?Object} */
+    this.attributes_;
+
+    /** @private {string} */
+    this.frameId_;
+
+    /** @private {string} */
+    this.frameUrl_;
+
+    /** @private {string} */
+    this.videoId_;
+
+    /** @private {string} */
+    this.header_;
+
+    /** @private {element} */
+    this.container_;
+
+    /** @private {string} */
+    this.placeholderUrl_;
+
+    /** @private {element} */
+    this.iframe_;
+
+    /** @private {!Object} */
+    this.contentWindow_;
+
+    /** @private {number} */
+    this.position_;
+
+    /** @private {Array<Array<number>>} */
+    this.playedRanges_;
+
+    /** @private {Object} */
+    this.metadata_;
+
+    /** @private {!Window} */
+    this.win;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.win.addEventListener('message', e => {
+      if (typeof e.data === 'string' && e.data.startsWith(this.header_)) {
+        const message = e.data.replace(this.header_, '');
+
+        this.messageListeners_.forEach(listener => {
+          listener(message);
+        });
+      }
+    });
+
+    this.attributes_ = this.parseAttributes_();
+
+    this.frameId_ = this.attributes_.id || `${Math.random() * 10e17}`;
+    this.frameUrl_ = new URL('https://std.wpcdn.pl/wpjslib/AMP-270-init-iframe/playerComponentFrame.html');
+    this.frameUrl_.searchParams.set('frameId', this.frameId_);
+    this.frameUrl_.searchParams.set('debug', 'ampPlayerComponent');
+
+    if (this.attributes_.url) {
+      this.videoId_ = /mid=(\d*)/g.exec(this.attributes_.url)[1];
+    } else {
+      this.videoId_ = this.attributes_.clip;
+    }
+
+    if (!this.videoId_) {
+      throw new Error('No clip specified');
+    }
+    this.header_ = `WP.AMP.PLAYER.${this.frameId_}.`;
+
+    this.registerAction('showControls', () => { this.showControls(); });
+    this.registerAction('hideControls', () => { this.hideControls(); });
+    this.registerAction('getMetadata', () => { this.getMetadata(); });
+
+    this.container_ = this.win.document.createElement('div');
+    this.element_.appendChild(this.container_);
+
+    this.placeholderUrl_ = this.attributes_.screenshot;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  isInteractive() {
+    return true;
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    placeholder.setAttribute('placeholder', 'true');
+
+    const image = this.win.document.createElement('amp-img');
+    image.setAttribute('layout', 'fill');
+    if (this.placeholderUrl_) {
+      image.setAttribute('src', this.placeholderUrl_);
+    }
+
+    placeholder.appendChild(image);
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const that = this;
```

Comment:
`that` is not necessary since you're using function arrows, which are permanently scoped to their outer `this` context. The compiler handles context binding for ES5, so you're good to remove `that` here.